### PR TITLE
Migrate to maven and last gluon mobile plugin

### PR DESCRIPTION
**Pull request by alexdeveng**
Migrate project to maven in order to use the last gluon plugin for maven

### CODE REVIEW

**Comment by jguitart** [comment]
Fixed android and ios build
Added README instructions to build

File: DevoxxClientCommon/pom.xml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    <name>DevoxxClientCommon</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.openjfx</groupId>
+            <artifactId>javafx-controls</artifactId>
+            <version>${javafx.version}</version>
+        </dependency>
+
+        <!--        attach-->
```
Comment (abelsromero) [inline DevoxxClientCommon/pom.xml:23]: Is this comment worth keeping?

File: DevoxxClientMobile/pom.xml [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+                        <item>loggingON.properties</item>
+                        <item>loggingOFF.properties</item>
+                    </resourcesList>
+                    <reflectionList>
+                        <list>com.devoxx.DevoxxDrawerPresenter</list>
+                        <list>com.devoxx.util.DevoxxNotifications</list>
+                        <list>com.devoxx.util.DevoxxSearch</list>
+                        <list>com.devoxx.views.AboutPresenter</list>
+                        <list>com.devoxx.views.AttendeeBadgePresenter</list>
+<!--                        <list>com.devoxx.views.AuthenticatePresenter</list>-->
```
Comment (abelsromero) [inline DevoxxClientMobile/pom.xml]: Can this be removed or is relevant?

File: DevoxxClientMobile/pom.xml [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+                        <list>com.devoxx.model.SessionType</list>
+                        <list>com.devoxx.model.Speaker</list>
+                        <list>com.devoxx.model.Sponsor</list>
+                        <list>com.devoxx.model.SponsorBadge</list>
+                        <list>com.devoxx.model.Tag</list>
+                        <list>com.devoxx.model.Talk</list>
+                        <list>com.devoxx.model.TalkSpeaker</list>
+                        <list>com.devoxx.model.Tenant</list>
+                        <list>com.devoxx.model.Track</list>
+<!--                        <list>com.devoxx.model.User</list>-->
```
Comment (abelsromero) [inline DevoxxClientMobile/pom.xml]: Can this be removed or is relevant?

File: README.md [hunk 4]
```diff
@@ -37,11 +37,135 @@ Before you clone and run the project, it is advisable to go through the short tu
...
 ```
 
+
+## Building
+
+## Common build steps
+
+Both Android and ios builds have a common requisite to build: uses graalvm.
+You will need to have gluon graalvm own distribution in order to generate a binary package to install or distribute.
+You can download latest graalvm distribution from [here](https://github.com/gluonhq/graal/releases/latest)
```
Comment (abelsromero) [inline README.md:50]: We can suggest sdkman too, maybe it's easier?

File: README.md [hunk 5]
```diff
@@ -37,11 +37,135 @@ Before you clone and run the project, it is advisable to go through the short tu
...
+export GRAALVM_HOME=/path/to/graalvm-svm-java17-darwin-gluon-22.0.0.3-Final/Contents/Home
+```
+
+### Android
+
+#### Prerequisites
+
+Android packages only can be built in linux operating system (or WSL2 in Windows, but not in Mac).
+
+In addition Android packages only can be built to architecture aarch64, not compatible with ARM mobiles.
```
Comment (abelsromero) [inline README.md:66]: Isn't `aarch64` == ARM 64 bits?

File: README.md [hunk 6]
```diff
@@ -37,11 +37,135 @@ Before you clone and run the project, it is advisable to go through the short tu
...
+- Gluonfx graalvm distribution (https://github.com/gluonhq/graal/releases/latest)
+- gcc (6+) and ld (2.26+) packages
+- Android SDK and optionally Android NDK
+- "platform-tools","platforms;android-30","build-tools;30.0.2","ndk-bundle","extras;android;m2repository","extras;google;m2repository" android packages.
+
+#### Build
+
+In file DevoxxClientMobile/pom.xml change the line
+
+```xml
```
Comment (abelsromero) [inline README.md:79]: Most of my comments are opinionated, except this.
We should not requier manual steps. We don't need it current PR, but we should create a task (trello or GH issue here) to have maven profiles for this.

File: README.md [hunk 7]
```diff
@@ -37,11 +37,135 @@ Before you clone and run the project, it is advisable to go through the short tu
...
+    <gluonfxTarget>android</gluonfxTarget>
+```
+
+Use java sdk 17.
+Define variable GRAALVM_HOME to your gluonfx release installed in prerequisites.
+
+```
+    export GRAALVM_HOME=/path/to/graalvm-svm-java17-linux-gluon-22.0.0.3-Final
```
Comment (abelsromero) [inline README.md:93]: This config is not the same as in the intro. I'd suggest to remove the intro maybe and ensure this is correct?

File: README.md [hunk 8]
```diff
@@ -37,11 +37,135 @@ Before you clone and run the project, it is advisable to go through the short tu
...
+
+```
+    mvn -Pandroid gluonfx:nativerun
+```
+
+### iOS
+
 For more details about the project, please refer to our [Wiki](https://github.com/barcelonajug/JBCNConf-Mobile-App/wiki).
 
+This process will use xcode in background. For this reason you should create a new xcode project with the same bundle identifier than this project, and apply the right provisioning profile.
```
Comment (abelsromero) [inline README.md:133]: I assume this  is about the project property `iosProvisioningProfile`, but that's empty? How is should be informed?

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    <modules>
+        <module>DevoxxClientMobile</module>
+        <module>DevoxxClientCommon</module>
+        <module>DevoxxRemoteFunctions</module>
+        <module>JBCNConfRemoteFunctions</module>
+    </modules>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.release>11</maven.compiler.release>
```
Comment (abelsromero) [inline pom.xml]: Shouldn't it be 17 as per readme¿?

**Review by abelsromero — CHANGES_REQUESTED** [review]
Thank you all for the effort! I skimed throuhg it because I lack a lot of knowledge about the details, my main focus was on maintenance, specially the README.

File: README.md (hunk 4 above)
Comment (jguitart) [reply README.md:50]: I am not sure sdkman's version works properly.

File: README.md (hunk 5 above)
Comment (jguitart) [reply README.md:66]: Not sure. It is not arm, for sure.

File: README.md (hunk 6 above)
Comment (jguitart) [reply README.md:79]: i tried other forms to do this (profiles or maven parameters). Doesn't work

File: README.md (hunk 8 above)
Comment (jguitart) [reply README.md:133]: I'm not sure. xcode gets the provisioning profile associated in xcode project. I don't know how.

File: README.md (hunk 7 above)
Comment (jguitart) [reply README.md:93]: done

File: pom.xml (hunk 9 above)
Comment (jguitart) [reply pom.xml]: Done

File: DevoxxClientMobile/pom.xml (hunk 3 above)
Comment (jguitart) [reply DevoxxClientMobile/pom.xml]: Removed

File: DevoxxClientMobile/pom.xml (hunk 2 above)
Comment (jguitart) [reply DevoxxClientMobile/pom.xml]: Removed

File: DevoxxClientCommon/pom.xml (hunk 1 above)
Comment (jguitart) [reply DevoxxClientCommon/pom.xml:23]: Removed

**Pull request merged**